### PR TITLE
Lumen support, fix helpers test

### PR DESCRIPTION
**Pull request by deividaspetraitis**
Support for Lumen:
- Instead of injecting laravels application refactored to Illuminate\Container\Container that is extentend by both laravels and lumen applications.
- Lumen doesn't support boot() methods in providers. In lumens case namespaces and modules is loaded in register() method.
- ....

Fixed helper test.

In future there should be added code boilerplates specific for Lumen instead used Laravel`s. However this pull request is good starting point.

### CODE REVIEW

File: src/LaravelModulesServiceProvider.php [hunk 1]
```diff
@@ -43,20 +43,36 @@ public function register()
...
     {
-        $this->app->booted(function ($app) {
-            Stub::setBasePath(__DIR__ . '/Commands/stubs');
+        Stub::setBasePath(__DIR__ . '/Commands/stubs');
 
-            if ($app['modules']->config('stubs.enabled') === true) {
-                Stub::setBasePath($app['modules']->config('stubs.path'));
-            }
-        });
+        switch (get_class($this->app)) {
```
Comment (nWidart) [inline src/LaravelModulesServiceProvider.php]: Please avoid using a switch statement.

File: tests/HelpersTest.php [hunk 2]
```diff
@@ -30,6 +30,6 @@ public function tearDown()
     /** @test */
     public function it_finds_the_module_path()
     {
-        $this->assertTrue(str_contains(module_path('Blog'), 'modules/Blog'));
+        $this->assertTrue(str_contains(module_path('Blog'), 'Modules/Blog'));
```
Comment (nWidart) [inline tests/HelpersTest.php]: Why was this needed, tests are green as it was ?

**Comment by joshbrw** [comment]
Any chance that `switch` statement could be refactored out? Potentially something like `$this->isLumen()`?

File: src/Module.php [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 namespace Nwidart\Modules;
 
+use Illuminate\Container\Container;
 use Illuminate\Foundation\AliasLoader;
 use Illuminate\Foundation\Application;
```
Comment (nWidart) [inline src/Module.php]: If Application isn't used anymore, no need for import.

**Comment by nWidart** [comment]
I think a better way would be to create a service provider dedicated to Lumen only. That way people on Lumen can use that service provider, and the laravel users use the laravel service provider.

**Comment by deividaspetraitis** [comment]
@nWidart good idea. Agree with you about dedicated service provider for Lumen.

File: src/Module.php [hunk 4]
```diff
@@ -170,14 +170,12 @@ public function boot()
...
      * Register module's translation.
-     *
-     * @return void
      */
     protected function registerTranslation()
     {
         $lowerName = $this->getLowerName();
 
-        $langPath = $this->getPath() . "/Resources/lang";
+        $langPath = $this->getPath().'/Resources/lang';
```
Comment (nWidart) [inline src/Module.php]: This change should not have happened.

File: src/Module.php [hunk 5]
```diff
@@ -198,7 +196,7 @@ public function json($file = null)
         }
 
         return array_get($this->moduleJson, $file, function () use ($file) {
-            return $this->moduleJson[$file] = new Json($this->getPath() . '/' . $file, $this->app['files']);
+            return $this->moduleJson[$file] = new Json($this->getPath().'/'.$file, $this->app['files']);
```
Comment (nWidart) [inline src/Module.php]: This change should not have happened.

File: src/Module.php [hunk 6]
```diff
@@ -249,7 +247,7 @@ public function register()
      */
     protected function fireEvent($event)
     {
-        $this->app['events']->fire(sprintf('modules.%s.' . $event, $this->getLowerName()), [$this]);
+        $this->app['events']->fire(sprintf('modules.%s.'.$event, $this->getLowerName()), [$this]);
```
Comment (nWidart) [inline src/Module.php]: This change should not have happened.

File: src/Module.php [hunk 7]
```diff
@@ -279,7 +277,7 @@ protected function registerProviders()
     protected function registerFiles()
     {
         foreach ($this->get('files', []) as $file) {
-            include $this->path . '/' . $file;
+            include $this->path.'/'.$file;
```
Comment (nWidart) [inline src/Module.php]: This change should not have happened.

File: src/Module.php [hunk 8]
```diff
@@ -400,7 +398,7 @@ public function delete()
      */
     public function getExtraPath($path)
     {
-        return $this->getPath() . '/' . $path;
+        return $this->getPath().'/'.$path;
```
Comment (nWidart) [inline src/Module.php]: This change should not have happened.

**Comment by nWidart** [comment]
Please remake a new pull request. You are changing code style that should not have changed.

**Comment by deividaspetraitis** [comment]
That's weird. Not sure why but in other case GrumPHP is failing. 
screenshot_8

**Comment by nWidart** [comment]
Hm I see, that's strange indeed. Will check it out.

**Comment by nWidart** [comment]
aha, for phpcsfixer 2, the config file is .php_cs.dist, and did not update grumphp to use that.

**Comment by deividaspetraitis** [comment]
Rolled back these code style changes and opened new pull request: https://github.com/nWidart/laravel-modules/pull/275
